### PR TITLE
RTK query Implementation

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
### Description of a change:
- Implemented RTK query for fetching APIs

### Associated Trello Link(s):
- https://trello.com/c/nR1DuU43/19-rtk-query

### Developers Note:
- The main file to look upon is shared/redux/reducers/tasksApi.ts
- Please ignore this file shared/redux/reducers/todoSlice.ts, this will be used in next PR for undo and redo.
- All the other files have very minor changes that can be ignored.

### CODE REVIEW

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]
Please review your own PR first, figure out the issue and fix it asap

File: shared/redux/reducers/tasksApi.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { Todo } from "@/modules/todo/models";
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const tasksApi = createApi({
+  reducerPath: "tasksApi",
+  baseQuery: fetchBaseQuery({
+    baseUrl: "http://localhost:3000",
```
Comment (nur-zaman-sazim) [inline shared/redux/reducers/tasksApi.ts:7]: read base url from .env

File: shared/redux/reducers/todoSlice.ts
Comment (nur-zaman-sazim) [inline shared/redux/reducers/todoSlice.ts:1]: Remove Comments please

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

File: shared/redux/reducers/todoSlice.ts
Comment (mohammed-sharraf-sazim) [reply shared/redux/reducers/todoSlice.ts:1]: Comments removed

File: shared/redux/reducers/tasksApi.ts (hunk 1 above)
Comment (mohammed-sharraf-sazim) [reply shared/redux/reducers/tasksApi.ts]: .env file added

**Comment by mohammed-sharraf-sazim** [comment]
> Please review your own PR first, figure out the issue and fix it asap

Issues figured it out and fixed

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**